scripts/sonic-kdump-config

```diff
@@ -0,0 +1,616 @@
+                return int(lines[0][p+2:])
+    except:
+        pass
+    return 3
+
+## Read current value for USE_KDUMP in kdump config file
+#
+#  @return The integer value X from USE_KDUMP=X in /etc/default/kdump-tools
+def read_use_kdump():
+    (rc, lines, err_str) = run_command("grep 'USE_KDUMP=.*' %s | cut -d = -f 2" % kdump_cfg, use_shell=True);
+    if rc == 0 and type(lines) == list and len(lines) >= 1:
+        try:
+            return int(lines[0])
+        except Exception as e:
+            print('Error! Exception[%s] occured while reading from %s' %(str(e), kdump_cfg))
+            sys.exit(1)
+    else:
+        print_err("Unable to read USE_KDUMP from %s" % kdump_cfg)
+        sys.exit(1)
+
+## Rewrite value for USE_KDUMP in kdump config file /etc/default/kdump-tools
+#
+#  @param use_kdump 0 or 1
+def write_use_kdump(use_kdump):
+    (rc, lines, err_str) = run_command("/bin/sed -i -e 's/USE_KDUMP=.*/USE_KDUMP=%s/' %s" % (use_kdump, kdump_cfg), use_shell=False);
+    if rc == 0 and type(lines) == list and len(lines) == 0:
+        use_kdump_in_cfg = read_use_kdump()
+        if use_kdump_in_cfg != use_kdump:
+            print_err("Unable to write USE_KDUMP into %s" % kdump_cfg)
+            sys.exit(1)
+    else:
+        print_err("Error while writing USE_KDUMP into %s" % kdump_cfg)
+        sys.exit(1)
+
+## Read current value for KDUMP_NUM_DUMPS in kdump config file
+#
+#  @return The integer value X from KDUMP_NUM_DUMPS=X in /etc/default/kdump-tools
+def read_num_dumps():
+    (rc, lines, err_str) = run_command("grep '#*KDUMP_NUM_DUMPS=.*' %s | cut -d = -f 2" % kdump_cfg, use_shell=True);
+    if rc == 0 and type(lines) == list and len(lines) >= 1:
+        try:
+            return int(lines[0])
+        except Exception as e:
+            print_err('Error! Exception[%s] occured while reading from %s' %(str(e), kdump_cfg))
+            sys.exit(1)
+    else:
+        print_err("Unable to read KDUMP_NUM_DUMPS from %s" % kdump_cfg)
+        sys.exit(1)
+
+## Change the value for KDUMP_NUM_DUMPS in kdump config file /etc/default/kdump-tools
+#
+#  #param num_dumps Integer value for new value
+def write_num_dumps(num_dumps):
+    (rc, lines, err_str) = run_command("/bin/sed -i -e 's/#*KDUMP_NUM_DUMPS=.*/KDUMP_NUM_DUMPS=%d/' %s" % (num_dumps, kdump_cfg), use_shell=False);
+    if rc == 0 and type(lines) == list and len(lines) == 0:
+        num_dumps_in_cfg = read_num_dumps()
+        if num_dumps_in_cfg != num_dumps:
+            print_err("Unable to write KDUMP_NUM_DUMPS into %s" % kdump_cfg)
+            sys.exit(1)
+    else:
+        print_err("Error while writing KDUMP_NUM_DUMPS into %s" % kdump_cfg)
+        sys.exit(1)
+
+## Command: Enable kdump
+#
+#  @param verbose If True, the function will display a few additinal information
+#  @param no      If True, will not display reboot needed message
+#  @return        True is the grub configuration has changed, and False if it has not
+def cmd_kdump_enable(verbose, no):
```

Comment:
Could the parsing of grub.cfg and the logic be separated into different functions?
The idea would be facilitating the integration of other bootloaders than grub by adding an abstraction layer.

I would propose something like the following in the case of cmd_kdump_enable. Some other places in this file would need some changes too. If a cmd_function refers to grub rather than a function of bootloader then it's lacking genericity.
```python

bootloader = get_bootloader() # this would return Grub() or Aboot() based on autodetection
if bootloader is None:
   print("Feature not supported on this platform")
   return False
image = get_current_image()
old_crashkernel = bootloader.get_crash_kernel( image )
new_crashkernel = "crashkernel: ..."
if old_crashkernel != new_crashkernel:
   bootloader.set_crash_kernel( image, new_crashkernel )
```

Reply:
@olivier-singla , since the sonic support multiple boot loader, not just grub. We have aboot, now we are working on u-boot. The concern here is that current code does not provide good abstraction for the community to add support for other types of boot load. Can you simplify the code logic and define the api. So that others can provide their own implementation for their own bootloader?